### PR TITLE
fix: noHost for collectLocalMetric

### PR DESCRIPTION
**Pull request by jiuker**
fix noHost for collectLocalMetric

## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description

for this pr https://github.com/minio/minio/pull/18188.
After reproducing the scene, it was found that `len(opts.hosts) > 0` is never true for `mc admin scanner info --json ALIAS ` 
image

And some cases are `globalMinioAddr` is port only like `:9000`. So we will get this
```
MacBook-Pro minio % mc admin scanner info --json minio
{
 "hosts": [
  ":9000",
  ":9000"
 ],
 "aggregated": {
  "scanner": {
   "collected": "2024-04-10T19:17:38.195757+08:00",
   "current_cycle": 0,
   "current_started": "0001-01-01T00:00:00Z",
   "cycle_complete_times": null,
   "last_minute": {}
  }
 },
 "final": false
}
```
This pr  use `globalLocalNodeName` instead.
After that is like 
```
MacBook-Pro minio % mc admin scanner info --json minio
{
 "hosts": [
  "10.70.4.5:9000",
  "10.70.4.93:9000"
 ],
 "aggregated": {
  "scanner": {
   "collected": "2024-04-10T19:29:46.902466+08:00",
   "current_cycle": 0,
   "current_started": "0001-01-01T00:00:00Z",
   "cycle_complete_times": null,
   "last_minute": {}
  }
 },
 "final": false
}
```

The most case is (Not linux)
```
MacBook-Pro minio % mc support top net minio ...
    SERVER      INTERFACE       RECEIVE TRANSMIT                                                                                                  
10.70.4.93:9000                  0 B/s   0 B/s                                                                                                   
10.70.4.5:9000                    ✗        ✗            10.70.4.5:9000: could not read "/proc": stat /proc: no such file or directory  (nicstats)
```

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Review by klauspost — APPROVED** [review]
LGTM

File: cmd/metrics-realtime.go [hunk 1]
```diff
@@ -41,6 +43,22 @@ func collectLocalMetrics(types madmin.MetricType, opts collectMetricsOpts) (m ma
 		return
 	}
 
+	byHostName := globalMinioAddr
```
Comment (shtripat) [inline cmd/metrics-realtime.go:46]:
```suggestion
	hostName := globalMinioAddr
```
otherwise changes look fine

Comment (jiuker) [reply cmd/metrics-realtime.go:46]: @harshavardhana Need a name here.

Comment (harshavardhana) [reply cmd/metrics-realtime.go:46]: byHostName is kind of an query variable name so in this context it's okay @shtripat

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**